app/src/test/java/co/nimblehq/compose/crypto/ui/screens/home/HomeViewModelTest.kt

```diff
@@ -0,0 +1,72 @@
+package co.nimblehq.compose.crypto.ui.screens.home
+
+import app.cash.turbine.test
+import co.nimblehq.compose.crypto.domain.usecase.GetMyCoinsUseCase
+import co.nimblehq.compose.crypto.test.MockUtil
+import co.nimblehq.compose.crypto.ui.screens.BaseViewModelTest
+import co.nimblehq.compose.crypto.ui.uimodel.toUiModel
+import io.kotest.matchers.shouldBe
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import org.junit.*
+
+@ExperimentalCoroutinesApi
+class HomeViewModelTest : BaseViewModelTest() {
+
+    private val mockGetMyCoinsUseCase = mockk<GetMyCoinsUseCase>()
+    private lateinit var viewModel: HomeViewModel
+
+    @Before
+    fun setUp() {
+        every { mockGetMyCoinsUseCase.execute(any()) } returns flowOf(MockUtil.myCoins)
+
+        Dispatchers.setMain(testDispatcher)
+    }
+
+    @After
+    fun tearDown() {
+        Dispatchers.resetMain()
+        testDispatcher.cleanupTestCoroutines()
+    }
+
+    @Test
+    fun `When getting my coin list successfully, it should return my coin list`() =
+        runBlockingTest {
+            testDispatcher.pauseDispatcher()
+            initViewModel()
+            val expected = MockUtil.myCoins.map { it.toUiModel() }
+            viewModel.output.myCoins.test {
+                testDispatcher.resumeDispatcher()
+                expectMostRecentItem() shouldBe expected
+                cancelAndIgnoreRemainingEvents()
+            }
+        }
+
+    @Test
+    fun `When getting my coin list failed, it should throw error`() =
+        runBlockingTest {
+            testDispatcher.pauseDispatcher()
```

Comment:
Question @luongvo Why do we need to call `pauseDispatcher()` before `initViewModel`, It doesn't clear when `cleanupTestCoroutines()` in `tearDown()` 🤔

Reply:
@Wadeewee to pause the vm dispatcher to mock failed response for GetMyCoinsUsecase. Because we're executing it at init block :nerd_face:

Reply:
Oh, I see, If that I think It would be better if we execute from `HomeScreen` regarding this [comment](https://github.com/nimblehq/jetpack-compose-crypto/pull/27#discussion_r9632202041) 🤔

Reply:
@Wadeewee it's not really better I think.
- As because we don't have a human interaction like Retry or pull down to refresh the my coins data so we don't need to create an input to load it from HomeScreen.
- Because we load it at init block, we need to use pause dispatcher to mock the test response. I don't think this is a bad thing as Test dispatcher is built for this case of testing. I don't think changing the production code to just make the test passed is a great idea.